src/core/xds/grpc/xds_cluster_parser.cc

```diff
@@ -84,40 +93,55 @@ constexpr absl::string_view kHttp11ProxyType =
     "envoy.extensions.transport_sockets.http_11_proxy.v3"
     ".Http11ProxyUpstreamTransport";
 
-CommonTlsContext UpstreamTlsContextParse(
+XdsClusterResource::UpstreamTlsContext UpstreamTlsContextParse(
     const XdsResourceType::DecodeContext& context,
     const XdsExtension& extension, ValidationErrors* errors) {
+  XdsClusterResource::UpstreamTlsContext upstream_tls_context;
   const absl::string_view* serialized_upstream_tls_context =
       std::get_if<absl::string_view>(&extension.value);
   if (serialized_upstream_tls_context == nullptr) {
     errors->AddError("can't decode UpstreamTlsContext");
     return {};
   }
-  const auto* upstream_tls_context =
+  const auto* upstream_tls_context_proto =
       envoy_extensions_transport_sockets_tls_v3_UpstreamTlsContext_parse(
           serialized_upstream_tls_context->data(),
           serialized_upstream_tls_context->size(), context.arena);
-  if (upstream_tls_context == nullptr) {
+  if (upstream_tls_context_proto == nullptr) {
     errors->AddError("can't decode UpstreamTlsContext");
     return {};
   }
+  if (XdsSniEnabled()) {
+    upstream_tls_context.sni = UpbStringToStdString(
```

Comment:
Please add tests in xds_cluster_resource_type_test.cc that cover parsing of these new fields.